File: 3289-the-two-sneaky-numbers-of-digitville/3289-the-two-sneaky-numbers-of-digitville.py
class Solution:
    def getSneakyNumbers(self, nums: List[int]) -> List[int]:
        ans = []

        for i in range(len(nums)):
            num = nums[i] - 1000 if nums[i] >= 1000 else nums[i]
            
            if nums[num] >= 1000:   # visited
                ans.append(num)
            else:
                nums[num] += 1000

        return ans

File: 3289-the-two-sneaky-numbers-of-digitville/3289-the-two-sneaky-numbers-of-digitville.cs
public class Solution {
    public int[] GetSneakyNumbers(int[] nums) {
        var ans = new int[2];
        var ansIdx = 0;

        for(var i=0; i<nums.Length; i++){
            var num = nums[i] >= 101 ? nums[i] - 101 : nums[i];
            
            if(nums[num] >= 101)   // visited
                ans[ansIdx++] = num;
            else
                nums[num] += 101;
        }

        return ans;
    }
}